Sources/Host/POSIX/PTrace.cpp

```diff
@@ -64,7 +64,13 @@ ErrorCode PTrace::attach(ProcessId pid) {
 
   DS2LOG(Debug, "attaching to pid %" PRIu64, (uint64_t)pid);
 
-  if (wrapPtrace(PTCMD(ATTACH), pid, nullptr, nullptr) < 0)
+#if defined(OS_LINUX) || defined(OS_FREEBSD)
+  auto cmd = PTCMD(ATTACH);
+#elif defined(OS_DARWIN)
+  auto cmd = PT_ATTACHEXC;
```

Comment:
Ah thanks for checking.

On Sun, May 8, 2016 at 8:35 PM, kokarez <notifications@github.com> wrote:

> In Sources/Host/POSIX/PTrace.cpp
> <https://github.com/facebook/ds2/pull/327#discussion_r62447649>:
>
> > @@ -64,7 +64,13 @@ ErrorCode PTrace::attach(ProcessId pid) {
> >
> >    DS2LOG(Debug, "attaching to pid %" PRIu64, (uint64_t)pid);
> >
> > -  if (wrapPtrace(PTCMD(ATTACH), pid, nullptr, nullptr) < 0)
> > +#if defined(OS_LINUX) || defined(OS_FREEBSD)
> > +  auto cmd = PTCMD(ATTACH);
> > +#elif defined(OS_DARWIN)
> > +  auto cmd = PT_ATTACHEXC;
>
> Yep. It work for me too, just weird that doesn't appear in the man.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/facebook/ds2/pull/327/files/c78509759aabbfa22365429f4d084bb2d370d7a5..3fd95111772021af410e7cbf5268ace0aec72410#r62447649>
>



-- 
Stephane Sezer